update/Pos/Update/Poll/PollState.hs

```diff
@@ -72,10 +73,16 @@ modifyPollState PollModifier {..} PollState {..} =
     PollState (modifyHashMap pmBVs _psBlockVersions)
               (fromMaybe _psAdoptedBV pmAdoptedBVFull)
               (fromMaybe _psEpochProposers pmEpochProposers)
-              (modifyHashMap pmConfirmed _psConfirmedBVs)
+              (modifyHashMap pmConfirmed _psConfirmedANs)
               (modifyHashMap pmConfirmedProps _psConfirmedProposals)
               (modifyHashMap pmActiveProps _psActiveProposals)
-              _psActivePropsIdx
+              newActivePropsIdx
               (fromMaybe _psSlottingData pmSlottingData)
               _psFullRichmenData
               _psIssuersStakes
+  where
+    (dels, ins) =
+        (HS.fromList . deletions) &&& (HS.fromList . map fst . insertions) $ pmActiveProps
+    newActivePropsIdx =
+        (fmap removeAndAddProps . (`HM.difference` pmDelActivePropsIdx)) _psActivePropsIdx
+    removeAndAddProps hashset = HS.union ins . (flip HS.difference dels) $ hashset
```

Comment:
There are at least two logical errors here.
1. I don't understand this `HM.difference pmDelActivePropsIdx`. Suppose there is we removed proposal with `id1` for application `a` but didn't remove proposal with `id2` for this application. If there are both `id1` and `id2`, both of them will be removed even if we remove only `id1`, for example.
2. Insertions are wrong too. There is `HS.union ins` which is applied to each set from `_psActivePropsIdx`. But `ins` contains all insertions, while each set from `_psActivePropsIdx` corresponds to particular application name. Only update proposals with application name `a` should be inserted into `_psActivePropsIdx . at a`.